### ISSUE TITLE
Radius not respected with systrays

### ISSUE THREAD

**Issue by sebastiencs** [polybar/polybar#651]
Hi,

When I set the radius and a tray icon is inserted it remove the radius:
![radius](https://user-images.githubusercontent.com/5273820/28198931-4ebf3590-6894-11e7-9c2d-1b96aec88a9c.gif)

**Comment by jaagr** [comment]
Hi,

The reason it does this is because the tray is actually another window on top of the bar window. To get around this you could offset the tray by the radius value, for example:

~~~ dosini
[bar/mybar]
radius = 5.0
tray-offset-x = ${self.radius}
~~~

**Comment by patrick96** [comment]
This actually doesn't work, even when the tray is offset, it leaves a black box (which is square) where it would be without offset

**Comment by sebastiencs** [comment]
I confirm, it doesn't work, even with the offset:
![polybar](https://user-images.githubusercontent.com/5273820/28212017-2512bc1a-68d2-11e7-9025-4d8204ea7446.jpg)

**Comment by PixelGGamer** [comment]
![screenshot from 2017-07-24 09-36-29](https://user-images.githubusercontent.com/7670940/28512863-f0ced892-7053-11e7-9031-0dfc5a2d4ef0.png)
The only way for now is setting it to center, which is well, ugly. Looking forward to solution

**Comment by karambaq** [comment]
+1

**Comment by tasmo** [comment]
One solution could be handle the tray as a module which made it possible to move the tray in its position before an other module.

**Comment by ReiLucio** [comment]
The only workaround i found was this, works fine only if your systray width is static... but rice comes first :

`padding-right = 13
`
`tray-detached = true
`
`tray-offset-x = -4
`
![selection_999 019](https://user-images.githubusercontent.com/32798365/42407767-02c8c8cc-8198-11e8-9895-665dad084692.png)

**Comment by gitneeraj** [comment]
> The only workaround i found was this, works fine only if your systray width is static... but rice comes first :
> 
> `padding-right = 13 `
> `tray-detached = true `
> `tray-offset-x = -4 `
> ![selection_999 021](https://user-images.githubusercontent.com/32798365/42408042-4de2d0fa-819d-11e8-9978-a94d9691faff.png)

Does not work if more tray icons appear. Like I have slack, zoom, nm etc.
Initially works fine but when dynamically icons are added, they overlap. :(

**Comment by AlecsFerra** [comment]
news?

**Comment by Fable-git** [comment]
I've had success by setting the tray-position to center and offsetting it to where I want it in the bar. Seems to be better to put it on the left so that it extends into your polybar.

;Somewhere in your bar
Padding-left = 40; if you want to have things to the right of your tray

tray-position = center
tray-detatched = true
tray-offset-x = -999;big number, depends

![polybar](https://user-images.githubusercontent.com/54632731/84599642-7c50d180-ae41-11ea-9a04-bc844fb30284.png)

also, if you have trouble with the systray rectangle still poking out a little bit, you can always lightly change the tray-scale.

**Comment by vivekascoder** [comment]
```
modules-left = i3
modules-right = pulseaudio cpu memory date temperature filesystem battery
module-margin = 2 
padding-right = 10
tray-position = right
tray-detached = true
;tray-maxsize = 16
tray-offset-x = -20 
tray-padding = 4 
;tray-transparent = false
```
This worked for me,

**Comment by tobiabocchi** [comment]
Same problem here, I can get it to work by setting the tray to detached but it's not an optimal solution.

**Comment by patri9ck** [comment]
Are there any plans for this getting fixed in near future so we can have a tray bar and rounded corners in a proper way? Maybe the tray bar should become a module.

**Comment by blackCauldron7** [comment]
please fix this.

**Comment by gsaker** [comment]
Same issue here, would be great if this could be fixed.

**Comment by ItzIlhxm** [comment]
please fix this

**Comment by Fxzzi** [comment]
bump, would love if this was fixed

**Comment by fP4rad0x** [comment]
same problem here

**Comment by SIMULATAN** [comment]
@fP4rad0x I believe this may be fixed by #2609, please look into that.

**Comment by Sewdohe** [comment]
Would also love to see this fix implemented!

**Comment by outp1** [comment]
oh... same problem

**Comment by patrick96** [comment]
With the new system tray module this will be "fixed" (#2689). The squares holding the tray icons will likely never be rounded, but now you can position the tray just like any module and thus also apply padding so that the tray icons are far enough away to not obstruct the rounded corners.
